monai/apps/auto3dseg/ensemble_builder.py

```diff
@@ -327,3 +339,244 @@ def get_ensemble(self):
+        self.work_dir = work_dir
+        self.num_fold = num_fold
+        self.ensemble_method_name = ensemble_method_name
+        self.mgpu = mgpu
+        self.kwargs = kwargs
+        self.rank = 0
+        self.world_size = 1
+        self.device_setting: dict[str, int | str] = {
+            "CUDA_VISIBLE_DEVICES": ",".join([str(x) for x in range(torch.cuda.device_count())]),
+            "n_devices": torch.cuda.device_count(),
+            "NUM_NODES": int(os.environ.get("NUM_NODES", 1)),
+            "MN_START_METHOD": os.environ.get("MN_START_METHOD", "bcprun"),
+            "CMD_PREFIX": os.environ.get("CMD_PREFIX"),  # type: ignore
+        }
+
+    def set_ensemble_method(self, ensemble_method_name: str = "AlgoEnsembleBestByFold", **kwargs: Any) -> None:
+        """
+        Set the bundle ensemble method
+
+        Args:
+            ensemble_method_name: the name of the ensemble method. Only two methods are supported "AlgoEnsembleBestN"
+                and "AlgoEnsembleBestByFold".
+            kwargs: the keyword arguments used to define the ensemble method. Currently only ``n_best`` for
+                ``AlgoEnsembleBestN`` is supported.
+
+        """
+        self.ensemble_method_name = look_up_option(
+            ensemble_method_name, supported=["AlgoEnsembleBestN", "AlgoEnsembleBestByFold"]
+        )
+        if self.ensemble_method_name == "AlgoEnsembleBestN":
+            n_best = kwargs.pop("n_best", False) or 2
+            self.ensemble_method = AlgoEnsembleBestN(n_best=n_best)
+        elif self.ensemble_method_name == "AlgoEnsembleBestByFold":
+            self.ensemble_method = AlgoEnsembleBestByFold(n_fold=self.num_fold)  # type: ignore
+        else:
+            raise NotImplementedError(f"Ensemble method {self.ensemble_method_name} is not implemented.")
+
+    def set_image_save_transform(self, **kwargs):
+        """
+        Set the ensemble output transform.
+
+        Args:
+            kwargs: image writing parameters for the ensemble inference. The kwargs format follows SaveImage
+                transform. For more information, check https://docs.monai.io/en/stable/transforms.html#saveimage .
+
+        """
+
+        if "output_dir" in kwargs:
+            output_dir = kwargs.pop("output_dir")
+        else:
+            output_dir = os.path.join(self.work_dir, "ensemble_output")
+            if self.rank == 0:
+                logger.info(f"The output_dir is not specified. {output_dir} will be used to save ensemble predictions")
+
+        if not os.path.isdir(output_dir):
+            os.makedirs(output_dir)
+            if self.rank == 0:
+                logger.info(f"Directory {output_dir} is created to save ensemble predictions")
+
+        self.output_dir = output_dir
+        output_postfix = kwargs.pop("output_postfix", "ensemble")
+        output_dtype = kwargs.pop("output_dtype", "$np.uint8")
+        resample = kwargs.pop("resample", False)
+
+        self.save_image = {
```

Comment:
cannot use SaveImage here because it's not compatible with `yaml.dump(config)` https://github.com/Project-MONAI/research-contributions/blob/8b10ad5b04d60835e1a2bd16a49abc378045b3f8/auto3dseg/algorithm_templates/segresnet/scripts/segmenter.py#L523